sdk/platform/http_client/curl/inc/curl_client.hpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <curl/curl.h>
+#include <http/http.hpp>
+
+class CurlClient
+{
+private:
+  azure::core::http::Request& _request;
+  CURL* _p_curl;
+
+  // setHeaders()
+  CURLcode setUrl()
+  {
+    return curl_easy_setopt(_p_curl, CURLOPT_URL, this->_request.getEncodedUrl().c_str());
+  }
+  CURLcode perform()
+  {
+    auto settingUp = setUrl();
+    if (settingUp != CURLE_OK)
+    {
+      return settingUp;
+    }
+    return curl_easy_perform(_p_curl);
+  }
+
+public:
+  CurlClient(azure::core::http::Request& request) : _request(request)
+  {
+    curl_global_init(CURL_GLOBAL_DEFAULT);
```

Comment:
The [documentation](https://curl.haxx.se/libcurl/c/curl_global_init.html) says **This function is not thread safe**. Would it be problematic if two threads initiate HTTP requests at the same time?

Reply:
actually I think we don't really need this at all. I will remove it.